src/Core/Commit.js

```diff
@@ -0,0 +1,363 @@
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ */
+function commitPromise (element, snapshot) {
+	snapshot.type.then(function (value) {
+		if (!element.DOM)
+			return
+
+		if (element.id === SharedElementPromise)
+			reconcileChildren(element, elementFragment(commitElement(value)))
+		else
+			reconcileElement(element, commitElement(value))
+	}).catch(function (e) {
+		errorBoundary(element, e, SharedSiteAsync+':'+SharedSiteRender, SharedErrorActive)
+	})
+}
+
+/**
+ * @param {Element} element
+ * @param {List} children
+ * @param {Element} host
+ * @param {number} signature
+ * @param {number} mode
+ */
+function commitChildren (element, children, host, signature, mode) {
+	var length = children.length
+	var sibling = children.next
+	var next = sibling
+
+	while (length-- > 0) {
+		if (next.DOM) {
+			sibling = next
+			children.insert(next = merge(new Element(SharedElementNode), next), sibling)
+			children.remove(sibling)
+		}
+
+		commitMount(next, element, element, host, signature, mode)
+		next = next.next
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} sibling
+ * @param {Element} parent
+ * @param {Element} host
+ * @param {number} signature
+ * @param {number} mode
+ */
+function commitMount (element, sibling, parent, host, signature, mode) {
+	element.host = host
+	element.parent = parent
+	element.context = host.context
+
+ 	switch (element.id) {
+ 		case SharedElementComponent:
+ 			element.work = SharedWorkTask
+ 			
+ 			commitMount(componentMount(element), sibling, parent, element, signature, mode)
+ 			element.DOM = element.children.DOM
+
+ 			if (element.ref)
+ 				commitReference(element, element.ref, SharedReferenceAssign)
+ 			
+ 			if (element.owner[SharedComponentDidMount])
+ 				getLifecycleMount(element, SharedComponentDidMount)
+
+ 			element.work = SharedWorkSync
+ 			return
+ 		case SharedElementPortal:
+ 			element.DOM = DOM(element.type)
+ 			break
+ 		case SharedElementPromise:
+ 			commitPromise(element, element)
+ 		case SharedElementFragment:
+ 			element.DOM = DOM(DOMTarget(parent))
+ 			break
+ 		case SharedElementNode:
+ 			element.xmlns = DOMType(element.type, parent.xmlns)
+ 		case SharedElementText:
+ 			switch (mode) {
+ 				case SharedMountClone:
+ 					if (element.DOM = DOMFind(element, parent))
+	 					break
+ 				default:
+ 					element.DOM = commitDOM(element)
+ 					
+ 					if (signature < SharedMountInsert)
+ 						commitAppend(element, parent)
+ 					else
+ 						commitInsert(element, sibling, parent)
+ 			}
+
+ 			if (element.id === SharedElementText)
+ 				return
+ 	}
+
+ 	commitChildren(element, element.children, host, SharedMountAppend, mode)
+ 	commitProperties(element, element.props, SharedPropsMount)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} parent
+ * @param {number} signature
+ */
+function commitUnmount (element, parent, signature) {
+	if (element.id === SharedElementComponent)
```

Comment:
Shouldn't this be `element.id === SharedElementComponent || element.id === SharedElementNode`?

Reply:
Only Component Elements are treated by `componentUnmount`.